evalml/pipelines/component_graph.py

```diff
@@ -97,6 +104,20 @@ def check_all_inputs_have_correct_syntax(edge):
                     "All edges must be specified as either an input feature ('X'/.x) or input target ('y'/.y)."
                 )
 
+            target_inputs = [
+                component
+                for component in component_inputs
+                if (component.endswith(".y"))
+            ]
+            if target_inputs:
+                target_component_name = target_inputs[0][:-2]
+                target_component_class = self.get_component(target_component_name)
+                if (
+                    not issubclass(target_component_class, TargetTransformer)
```

Comment:
Why not just use `not target_component_class.modifies_target`? All TargetTransformers modify the target by definition. 

In fact, I think we can get rid of the TargetTransformer abstraction in the future. I added that because at the time our component graph would allow you to specify `.y` for a component that did not modify the target so the component graph `inverse_transform` could error out. I will file an issue.

Reply:
I think you're absolutely right--thanks for filing the issue! 😁